frontend/app.js

```diff
@@ -505,6 +505,29 @@
         });
     });
 
+    app.run(function featureToggleInterceptor(FeatureToggleService, MapStateToFeatureService, $transitions, STATES) {
+        
+        $transitions.onBefore({
+            to: function(state) {
+                const stateName = state.name;
+                return MapStateToFeatureService.containsFeature(stateName);
+            }
+        }, function (transition) {
+            const targetStateName = transition.to().name;
+
+            return FeatureToggleService.isEnabled(MapStateToFeatureService.getFeatureByState(targetStateName)).then(function(enabled) {
+                if (enabled) {
+                    return transition;
+                } else {
+                    return transition.router.stateService.target(STATES.ERROR, {
+                        "msg": "Desculpa! Este recurso ainda não está disponível.",
+                        "status": "500"
```

Comment:
I think it is not a typical 500 error, which means server error, change to a status that tells more about the specific error